src/mono/mono/metadata/metadata.c

```diff
@@ -1003,6 +1003,21 @@ mono_metadata_table_bounds_check_slow (MonoImage *image, int table_index, int to
         return mono_metadata_update_table_bounds_check (image, table_index, token_index);
 }
 
+static void
+compute_column_offsets (guint32 bitfield, guint8 column_offsets[MONO_TABLE_INFO_MAX_COLUMNS])
+{
+	int offset = 0, c = mono_metadata_table_count (bitfield);
+	for (int i = 0; i < MONO_TABLE_INFO_MAX_COLUMNS; i++) {
+		if (i >= c) {
+			column_offsets[i] = 0xFFu;
+		} else {
+			int size = mono_metadata_table_size (bitfield, i);
+			column_offsets[i] = (guint8)offset;
+			offset += size;
+		}
+	}
```

Comment:
probably doesn't matter, but why not `memset` and then overwrite just the fields from 0 to c-1?  

Also why do we need to initialize the out of range fields to a non-zero value?

Reply:
0 is a valid column offset, so the out of range ones are an invalid offset which we can assert against
could do memset, yeah. will change it if you want